keps/sig-cli/kustomize-file-processing-integration.md

```diff
@@ -0,0 +1,101 @@
+---
+title: Integrate Kustomize into cli-runtime
+authors:
+ - "@pwittrock"
+owning-sig: sig-cli
+participating-sigs:
+ - sig-cli
+reviewers:
+ - "@liggitt"
+ - "@seans3"
+ - "@soltysh"
+ - "@monopole"
+approvers:
+ - "@liggitt"
+ - "@seans3"
+ - "@soltysh"
+ - "@monopole"
+editors:
+ - "@pwittrock"
+creation-date: 2019-01-17
+last-updated: 2019-01-17
+status: provisional
+see-also:
+ - "kustomize-subcommand-integration.md"
+replaces:
+superseded-by:
+ - n/a
+---
+
+# Enable kustomize subcommand in kubectl
+
+## Table of Contents
+* [Table of Contents](#table-of-contents)
+* [Summary](#summary)
+* [Motivation](#motivation)
+  * [Goals](#goals)
+  * [Non-Goals](#non-goals)
+* [Proposal](#proposal)
+  * [Implementation Details/Notes/Constraints](#implementation-detailsnotesconstraints)
+  * [Risks and Mitigations](#risks-and-mitigations)
+* [Graduation Criteria](#graduation-criteria)
+* [Implementation History](#implementation-history)
+* [Alternatives](#alternatives)
+
+[Tools for generating]: https://github.com/ekalinin/github-markdown-toc
+
+## Summary
+
+This is a follow up to [KEP Kustomize Subcommand Integration](kustomize-subcommand-integration.md)
+
+[Kustomize](https://github.com/kubernetes-sigs/kustomize) is a subcommand of kubectl.  However it is
+not tightly integrated into the file processing libraries, creating user friction around error
+handling and messaging.
+
+## Motivation
+
+- It is capable of removing friction that requires deeper integration - such as producing errors referencing line
+  numbers of the original files (rather than the output files) and providing the right error code if kustomization
+  fails.
+- It is more consistent with UX workflow with other commands and flags
+- It has a cleaner and simpler UX than pipes
+- It is clear which commands it should be used with - apply, get, delete, etc.
+
+### Goals
+
+- Provide a clean and integrated user experience when working with files from kubectl.
+- Provide consistent UX across kubectl commands for working with kustomized applications.
+
+### Non-Goals
+
+## Proposal
+
+Integrate kustomize directly into libraries that enable file processing for cli-runtime (e.g. resource builder).
+Kubectl commands taking the common flags (`-f`, `--filename`, `-R`, `--recursive`) will support `kustomization.yaml`
+files.
+
+Cli-runtime will add the flags `-k, --kustomize=[]`, which will be registered along side the other file processing
```

Comment:
Quick question, I assume `-k` and `-f ` will be exclusive? If so, maybe it's worth adding that here.

Reply:
Hm.  Good question.  We could start that way and open it up in the future.

Reply:
Yeah, exclusive seems the best to start with, it'll leave us more room for change than the other way around.